lib/web_git/diff.rb

```diff
@@ -0,0 +1,249 @@
+        ones = {}
+        files.each_with_index do |file, i|
+          file_content = ""
+          lines = file.split("\n").drop(4)
+          lines.each do |line|
+            if !line.match?(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/)
+              if line.first != "+"
+              elsif line.first == "+"
+                line.slice!(0)
+                file_content += line + "\n"
+              end
+            else
+              file_content += "\n"
+            end
+          end
+          ones[filenames[i]] = file_content.chomp "\n"
+        end
+        ones
+      end
+
+      def self.last_to_html(diff)
+        left_hash = self.get_last_left(diff)
+        right_hash = self.get_last_right(diff)
+
+        html_output = '<link rel="stylesheet"' +
+          'href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/' +
+          'bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/' +
+          '1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">'
+        html_output += '<style>'
+        html_output += Diffy::CSS
+        html_output += '</style>'
+        html_output += '<div class="card">'
+        left_hash.keys.each do |file|
+          html_output += '<div class="file mb-4 p-3">'
+          html_output += '<h4>' + file + '</h4>'
+          html_output += Diffy::Diff.new(
+            left_hash[file],
+            right_hash[file],
+            :include_plus_and_minus_in_html => true
+          ).to_s(:html)
+          html_output += '</div>'
+        end
+        html_output += '</div>'
+        html_output
+      end
+
+      def self.diff_to_html(diff)
+        left_hash = self.get_each_left(diff)
+        right_hash = self.get_each_right(diff)
+        html_output = '<div style="overflow-y: scroll;height:400px">'
+        html_output += '<style>'
+        html_output += Diffy::CSS
+        html_output += '</style>'
+        html_output += '<div class="row mb-3">'
+        html_output += '<div class="col-md-12 offset" style="overflow-y: scroll;">'
+
+        left_hash.keys.each do |file|
+          html_output += '<div class="row text-center">
+            <div class="col-12">
+              <h4>'
+          html_output+= file.to_s
+          html_output += '</h4>
+            </div>
+          </div>
+            <div class="row mb-4">
+              <div class="col-6">'
+          html_output += Diffy::SplitDiff.new(
+            left_hash[file],
+            right_hash[file],
+            :format => :html
+            ).left
+
+          html_output +=
+              '</div>
+              <div class="col-6">'
+          html_output += Diffy::SplitDiff.new(
+            left_hash[file],
+            right_hash[file],
+            :format => :html
+          ).right
+
+          html_output += '</div></div>'
+        end
+        html_output += "</div>"
+        html_output += "</div>"
+        html_output += "</div>"
+        html_output
+
+      end
+  end
+end
```

Comment:
General comment for the file...

I noticed this CodeClimate suggestion and it caught my eye. Inside most of the class method definitions you call `self.[SOME OTHER CLASS METHOD]`. I think it'd be cleaner to eliminate the `self` calls and just call the methods (which you can do considering they're within the same scope).

Reply:
Nice catch! I think I did a find and replace on most of the methods which added some more `self`s then I meant to have.